pkg/kube/workload.go

```diff
@@ -278,7 +278,7 @@ func WaitOnDeploymentReady(ctx context.Context, kubeCli kubernetes.Interface, na
 func WaitOnDeploymentConfigReady(ctx context.Context, osCli osversioned.Interface, kubeCli kubernetes.Interface, namespace, name string) error {
 	return poll.Wait(ctx, func(ctx context.Context) (bool, error) {
 		ok, err := DeploymentConfigReady(ctx, osCli, kubeCli, namespace, name)
-		if apierrors.IsNotFound(errors.Cause(err)) {
+		if apierrors.IsNotFound(errkit.Unwrap(err)) {
```

Comment:
Not critical, but I wonder if we can unwrap an error to get a particular type. Would `errors.As` do that?

Reply:
In fact, under the hood, `IsNotFound` method is doing something like that. 

They don't have `NotFound` error type, instead they have `APIStatus` type. They are doing `errors.As` to this type and then they are checking `reason` and `code`.